### PR TITLE
config: validate online configurable thread pools (tikv#11159)

### LINKED ISSUE #11159: Support adjusting the pool size of raftstore poller threads dynamically
### Describe the feature you'd like:
A dynamic config modification is performed from TiKV-Control command line ([documentation](https://docs.pingcap.com/tidb/dev/tikv-control#modify-the-tikv-configuration-dynamically)). So expect to adjust the thread pool size in this way:
```sql
set config tikv `raftstore.apply-pool-size`=3
set config tikv `raftstore.store-pool-size`=3
```

### Technical proposal
- Add a `PoolStateBuilder` to `BatchSystem` to create a `PoolState` object to save what is needed to build a `Poller`；
- Define a `RefreshConfigRunner` to handle dynamic configuration updates;
- Define a `PoolController` to save the `PoolState` object and handle dynamically adjusting the pool size of the `Poller`;

### PR DESCRIPTION
**Pull request by ethercflow**
### What is changed and how it works?

Close  #11159  

What's Changed:

- validate online configurable thread pools 

### Check List 

Tests 

- Manual test
  - test with mysql client 
- No code

Side effects

- Performance regression
   No
- Breaking backward compatibility
   No

### Release note 

```release-note
None
```

### CODE REVIEW

File: src/config.rs [hunk 1]
```diff
@@ -1126,6 +1126,25 @@ impl DbConfig {
                 return Err("pipelined_write is not compatible with unordered_write".into());
             }
         }
+
+        // Since the following configuration supports online update, in order to
+        // prevent mistakenly inputting too large values, the max limit is made
+        // according to the cpu quota.
+        let cpu_num = SysQuota::cpu_cores_quota() as i32;
+        if self.max_background_jobs <= 0 || self.max_background_jobs > cpu_num {
```
Comment (5kbpers) [inline src/config.rs:1134]: I'm not sure if this limitation could fit in these config items. cc @Little-Wallace @Connor1996

Comment (Connor1996) [reply src/config.rs:1134]: lgtm

**Review by 5kbpers — APPROVED** [review]

**Review by Connor1996 — APPROVED** [review]

**Review by NingLin-P — APPROVED** [review]

**Pull request merged**